mailserver/mailserver.go

```diff
@@ -111,11 +90,22 @@ func (s *WMailServer) Init(shh *whisper.Whisper, config *params.WhisperConfig) e
 
 	// Open database in the last step in order not to init with error
 	// and leave the database open by accident.
-	database, err := db.Open(config.DataDir, nil)
-	if err != nil {
-		return fmt.Errorf("open DB: %s", err)
+	if config.DatabaseConfig.PGConfig.Enabled {
+		log.Info("Connecting to postgres database")
+		database, err := NewPostgresImpl(config)
```

Comment:
I would prefer something like `NewPostgresDB` and `NewLevelDB`, no need to explicitly state that these are implementations, I think it is obvious anyway.